ci/scripts/install_pandas.sh

```diff
@@ -28,7 +28,7 @@ pandas=$1
 
 if [ "${pandas}" = "master" ]; then
   conda install -q numpy
-  pip install https://github.com/pandas-dev/pandas/archive/master.tar.gz
+  pip install git+https://github.com/pandas-dev/pandas.git --no-build-isolation
```

Comment:
Nice!

Reply:
Apparently installing from the github archive gives no version number (you get something like "0+unknown" as the version), and when we import pandas in pyarrow, we have imports dependent on the pandas version, which was all failing.

Reply:
Sounds reasonable. Thanks for fixing it!